src/dataset/create_dataset.py

```diff
@@ -269,15 +171,16 @@ def get_embeddings_at_index(texts: str, batch_idx: int, batch_size: int = 200):
         rate_limit = 3500 / 60  # Maximum embeddings per second
 
         start = time.time()
-        self.embeddings = np.zeros((len(self.embedding_strings), LEN_EMBEDDINGS))
+        embedding_strings = [chunk_data['text'] for chunk_data in self.metadata]
+        self.embeddings = np.zeros((len(embedding_strings), LEN_EMBEDDINGS))
 
         with concurrent.futures.ThreadPoolExecutor() as executor:
             futures = [executor.submit(
```

Comment:
how about using [map](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.map) here? e.g.
```
results = executor.map(get_embeddings_at_index, chunks)
for (batch_idx, embeddings) in results:
    (...)
```
This assumes that there is a `chunks` iterable with the embeddings already chunked up